arch/arm/include/aarch32/cortex_m/tz.h

```diff
@@ -22,6 +22,7 @@
 
 #include <arm_cmse.h>
 #include <zephyr/types.h>
+#include <../arch/arm/include/aarch32/cortex_m/wrap_func.h>
```

Comment:
Why is this relative include path so ugly @oyvindronningstad ? could we add some include directory that simplifies this inclusion?

Reply:
```suggestion
#include <aarch32/cortex_m/wrap_func.h>
```

Reply:
let me know if that doesn't work.
What board and sample are you testing with ?

Reply:
@tejlmand it doesn't work for me.

Reply:
zephyr/tests/arch/arm/arm_wrap_func/src/main.c:8:10: fatal error: aarch32/cortex_m/wrap_func.h: No such file or directory
        8 | #include <aarch32/cortex_m/wrap_func.h>
          |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    compilation terminated.

Reply:
With board `nrf5340pdk_nrf5340_cpuapp`

Reply:
Could you provide some more details, cause building the `tests/arch/arm/arm_wrap_func` works here:
```
$ git diff
diff --git a/arch/arm/include/aarch32/cortex_m/tz.h b/arch/arm/include/aarch32/cortex_m/tz.h
index dcb50d9f00..467c1e0f2b 100644
--- a/arch/arm/include/aarch32/cortex_m/tz.h
+++ b/arch/arm/include/aarch32/cortex_m/tz.h
@@ -22,7 +22,7 @@
 
 #include <arm_cmse.h>
 #include <zephyr/types.h>
-#include <../arch/arm/include/aarch32/cortex_m/wrap_func.h>
+#include <aarch32/cortex_m/wrap_func.h>
 
 #ifdef __cplusplus
 extern "C" {

$ rm -rf build/; cmake -GNinja -DBOARD=nrf5340pdk_nrf5340_cpuapp -Bbuild  tests/arch/arm/arm_wrap_func
Including boilerplate (Zephyr repository): /projects/github/ncs/zephyr/cmake/app/boilerplate.cmake
-- Application: /projects/github/ncs/zephyr/tests/arch/arm/arm_wrap_func
-- Zephyr version: 2.3.99 (/projects/github/ncs/zephyr)
-- Found Python3: /usr/bin/python3.7 (found suitable exact version "3.7.5") found components:  Interpreter 
-- Found west: /home/tora/.local/bin/west (found suitable version "0.7.99", minimum required is "0.7.1")
-- Board: nrf5340pdk_nrf5340_cpuapp
-- Found toolchain: zephyr (/opt/zephyr-sdk-0.11.3)
-- Found dtc: /opt/zephyr-sdk-0.11.3/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /projects/github/ncs/zephyr/boards/arm/nrf5340dk_nrf5340/nrf5340pdk_nrf5340_cpuapp.dts
nrf5340pdk_nrf5340_cpuapp.dts.pre.tmp:55.42-67.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/flash-controller@39000: duplicate unit-address (also used in node /soc/peripheral@50000000/kmu@39000)
nrf5340pdk_nrf5340_cpuapp.dts.pre.tmp:329.19-335.3: Warning (unique_unit_address_if_enabled): /soc/peripheral@50000000/clock@5000: duplicate unit-address (also used in node /soc/peripheral@50000000/power@5000)
nrf5340pdk_nrf5340_cpuapp.dts.pre.tmp:577.31-578.5: Warning (unique_unit_address_if_enabled): /reserved-memory/image@20000000: duplicate unit-address (also used in node /reserved-memory/image_s@20000000)
  also defined at nrf5340pdk_nrf5340_cpuapp.dts.pre.tmp:597.14-599.3
-- Generated zephyr.dts: /projects/github/ncs/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /projects/github/ncs/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
Parsing /projects/github/ncs/zephyr/Kconfig
Loaded configuration '/projects/github/ncs/zephyr/boards/arm/nrf5340dk_nrf5340/nrf5340pdk_nrf5340_cpuapp_defconfig'
Merged configuration '/projects/github/ncs/zephyr/tests/arch/arm/arm_wrap_func/prj.conf'
Configuration saved to '/projects/github/ncs/zephyr/build/zephyr/.config'
Kconfig header saved to '/projects/github/ncs/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/zephyr-sdk-0.11.3/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Cache files will be written to: /home/tora/.cache/zephyr
-- Configuring done
-- Generating done
-- Build files have been written to: /projects/github/ncs/zephyr/build

$ ninja -Cbuild 
ninja: Entering directory `build'
[1/133] Preparing syscall dependency handling

[128/133] Linking C executable zephyr/zephyr_prebuilt.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       29640 B         1 MB      2.83%
            SRAM:        4920 B       448 KB      1.07%
        IDT_LIST:          56 B         2 KB      2.73%
[133/133] Linking C executable zephyr/zephyr.elf
```

Reply:
what is the complete compile command being invoked ?

Reply:
See you have this line in `main.c`
Please note that this header file is in `arch/arm/include/aarch32/cortex_m/tz.h`.

So we are talking about two different files here.

Reply:
Ah ok

Reply:
I did have problems in tz.h as well, but only in certain configurations...

Reply:
But that was with `#include <wrap_func.h>`. You fix seems to work fine in tz.h 👍

Reply:
Still fails in main.c.

    west build tests/arch/arm/arm_wrap_func/ -p

Reply:
The reason for that is because the `${ARCH_DIR}/${ARCH}/include` is not a standard include.

It's only used on targets that has special requirements, such as the kernel:
https://github.com/zephyrproject-rtos/zephyr/blob/master/kernel/CMakeLists.txt#L66

So what you need in the test app is:
```
 target_include_directories(app PRIVATE
  ${ARCH_DIR}/${ARCH}/include
)
```